### PR TITLE
[inductor] turn on loaf (for oss) by default

### PR DESCRIPTION
**Pull request by shunting314**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #162030

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @Lucaskabela

### CODE REVIEW

**Comment by shunting314** [comment]
Here is the benchmark result [link](https://hud.pytorch.org/benchmark/compilers?dashboard=torchinductor&startTime=Mon%2C%2001%20Sep%202025%2017%3A14%3A28%20GMT&stopTime=Mon%2C%2008%20Sep%202025%2017%3A14%3A28%20GMT&granularity=hour&mode=training&dtype=amp&deviceName=cuda%20(h100)&lBranch=gh/shunting314/215/head&lCommit=e4f4c17ee1709df3045a65b8cc4279ea0e3c9474&rBranch=main&rCommit=1f820de639c75a1562d3fb03f160439f853ae07b)

- Compilation time hit is about 2s
- the perf metric is mostly neutral. For the slowed model showed up on the dashboard, either the absolution execution time get smaller or I can not repro the slow down on my dev server.

Should this be a good state to land or we want to further optimize for compilation time? @jansel @eellison

**Review by eellison — APPROVED** [review]
Looks good but lets wait until https://github.com/pytorch/pytorch/pull/162316 is in

**Review by jansel — APPROVED** [review]
